src/olympia/lib/settings_base.py

```diff
@@ -1108,6 +1110,7 @@ def get_db_config(environ_var, atomic_requests=True):
 CSP_OBJECT_SRC = ("'none'",)
 
 CSP_SCRIPT_SRC = (
+    "'self'",
```

Comment:
Err, that's a deal breaker. For prod, we absolutely don't want to allow `self` as a `script-src` in our CSP, that's too dangerous.

Reply:
Another option, we can exclude `/api` or some subset of that from CSP... those URLs are not "really" relevant for CSP anyway right? Or are we worried about host spoofing code on specifically that subset.. can you explain in more detail the set of security vulnerabilities we are trying to defend against so it's easier to find a solution that meets the requirement?

Reply:
We're worried about any XSS on AMO domain, which would have typical wide consequences that XSS can have: stealing cookies, creating fake pages misleading users, etc.

This would be less likely on `/api/`... but still possible, in fact with swagger it's even more likely as it makes us host HTML content...

A strict CSP on all our responses is a non-negotiable requirement. If anything it should be stricter on `/api/` given that `/api/` shouldn't host anything that needs to load content (we gave up on it a while ago because it was low priority, but we did consider that at some point: https://github.com/mozilla/addons/issues/1855).

Reply:
I've got another workaround which is handling this file the way we handle javascript catalog files. Currently the "dev" version generating on demand is working, but the management command to statically generate for production is failing.. in our production nginx, are requests to /static that are not resolved by our static assets bucket forwarded to django? I guess not right. That would really simplify this because then I could just handle it dynamically every time and call it a day.

Reply:
No fallback no. It's a bit messy, it goes through a CDN config and then nginx, ideally it would be CDN + Google Cloud Storage instead (as is already the case for addons-frontend static files), so I would rather not rely on adding a fallback mechanism.

Reply:
I think I got it.. let's see. We have a per version url path based on /static/js/swagger/{version}.js. IN dev, this is rendered on the fly and during the build we pre generate each version in the static-build dir which should be collected and thus available in prod.

Reply:
Works.

Reply:
You still have 'self' allowed though ?

Reply:
Fixed. hopefully the verification still passes 🙃